core/src/main/java/org/pentaho/di/core/plugins/BasePluginType.java

```diff
@@ -650,9 +656,119 @@ protected URLClassLoader createUrlClassLoader( URL jarFileUrl, ClassLoader class
 
     urls.add( jarFileUrl );
 
-    return new KettleURLClassLoader( urls.toArray( new URL[urls.size()] ), classLoader );
+    KettleURLClassLoader urlClassLoader = new KettleURLClassLoader( urls.toArray( new URL[ urls.size() ] ), classLoader );
+    return processPluginClasspath( urlClassLoader, jarFileUrl, urls, classLoader );
+  }
+
+  /**
+   * Adds the entries defined in the classpath.properties located at the root of the plugin to the plugin classpath
+   */
+  private KettleURLClassLoader processPluginClasspath(
+    KettleURLClassLoader urlClassLoader, URL jarFileUrl, List<URL> urls, ClassLoader classLoader ) {
+    try {
+      String pluginRootFolderName = new File( URLDecoder.decode( jarFileUrl.getFile(), "UTF-8" ) ).getParent();
+      File pluginRootFolder = new File( pluginRootFolderName );
+      if( pluginRootFolder.exists() ) {
+        File classPathFile = new File( pluginRootFolder, "classpath.properties" );
```

Comment:
java constants for "classpath" and "classpath.properties" and the usage of these file names and properties should be in a javadoc comment somewhere in a public function